crates/rpc/rpc/src/eth/api/sign.rs

```diff
@@ -0,0 +1,29 @@
+//! Contains RPC handler implementations specific to sign endpoints
+
+use crate::{
+    eth::error::{EthApiError, EthResult},
+    EthApi,
+};
+use reth_primitives::{Address, Bytes, Signature};
+use reth_provider::{BlockProvider, EvmEnvProvider, StateProviderFactory};
+use reth_transaction_pool::TransactionPool;
+
+impl<Client, Pool, Network> EthApi<Client, Pool, Network>
+where
+    Pool: TransactionPool + 'static,
+    Client: BlockProvider + StateProviderFactory + EvmEnvProvider + 'static,
+    Network: 'static,
+{
+    pub(crate) async fn sign(&self, account: Address, message: Bytes) -> EthResult<Bytes> {
+        let signer = self
+            .inner
+            .signers
+            .iter()
+            .find(|signer| signer.is_signer_for(&account))
+            .ok_or(EthApiError::NoSigner)?;
+        let signature = signer.sign(account, &message).await.map_err(|_err| EthApiError::NoSigner)?;
+        let mut buff = vec![];
+        Signature::encode(&signature, &mut buff);
```

Comment:
the sign RPC responses are `hex(sig)` not hex(rlp(sig))

also the response should use the `v` value and not `odd_y_parity`

looks like we need additional helper functions for Signature.

this is what anvil returns for example:
https://github.com/gakonst/ethers-rs/blob/72428503cfb0855579d47f691b48b2d65ad8195b/ethers-core/src/types/signature.rs#L65-L70
which already has the v value set.

Reply:
Ok, so, I've done a couple of changes:
- The `hex(sig)`part is being done [here](https://github.com/paradigmxyz/reth/pull/1665/files#diff-e215e50528053cd9402e7ad953d55209aa4399f5f581236bed4d582970bd793eR103). This turns the signature into a [u8;65], [like in the file you linked](https://github.com/gakonst/ethers-rs/blob/72428503cfb0855579d47f691b48b2d65ad8195b/ethers-core/src/types/signature.rs#L223)
- I've also added a From<&Signature> to Bytes as to return a Result<Bytes> on the handler.

Let me know if this is correct or if there's something else that needs fixing ^^